### PR TITLE
Chore/Apple Auth email metadata

### PR DESCRIPTION
**Pull request by borut-t**
Added `JWTDecoder` util to decode JWT tokens.

### CODE REVIEW

File: Sources/Auth/Core/JWTDecoder.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+//
+//  Created by Borut Tomazin on 10/1/2023.
+//  Copyright © 2023 Povio Inc. All rights reserved.
+//
+
+import Foundation
+
+/// JWTDecoder for decoding JSON Web Tokens (JWT) tokens
+/// Inspired by https://github.com/auth0/jwt-decode
+public struct JWTDecoder {
```
Comment (gentique) [inline Sources/Auth/Core/JWTDecoder.swift:13]: We could write some tests for this implementation.

Comment (borut-t) [reply Sources/Auth/Core/JWTDecoder.swift:13]: Didn't get there yet. A separate PR is incoming.

File: Sources/Auth/Apple/AppleAuthProvider.swift [hunk 2]
```diff
@@ -96,9 +96,21 @@ extension AppleAuthProvider: ASAuthorizationControllerDelegate {
       // store userId for later
       Self.storage.set(credential.user, forKey: Self.userIdStorageKey)
       
+      // parse email and it's metadata
```
Comment (tonikocjan) [inline Sources/Auth/Apple/AppleAuthProvider.swift:99]: These and the next two comments feel a bit unnecessary.

File: Sources/Auth/Core/AuthProvider.swift [hunk 3]
```diff
@@ -31,3 +31,17 @@ public struct AuthProvider {
     case credentialsRevoked
   }
 }
+
+public extension AuthProvider.Response {
+  struct Email {
+    let address: String
+    let isPrivate: Bool?
+    let isVerified: Bool?
```
Comment (tonikocjan) [inline Sources/Auth/Core/AuthProvider.swift:39]: Are these properties supposed to be visible to the outside world? If yes, they need to be `public`.

**Review by tonikocjan — APPROVED** [review]

File: Sources/Auth/Apple/AppleAuthProvider.swift (hunk 2 above)
Comment (borut-t) [reply Sources/Auth/Apple/AppleAuthProvider.swift:99]: I think better more than less 🤷‍♂️

File: Sources/Auth/Core/AuthProvider.swift (hunk 3 above)
Comment (borut-t) [reply Sources/Auth/Core/AuthProvider.swift]: Visible yes. Pushed an [update](https://github.com/poviolabs/PovioKit/pull/236/commits/9b3b2f960c6525ba50db2a63f69caac620a7df76).

**Review by gentique — APPROVED** [review]

**Pull request merged**